static/js/settings.js

```diff
@@ -0,0 +1,118 @@
+    if (response.status !== 200) {
+        // TODO: Implement a proper error page
+        return
+    }
+
+    const responseData = await response.json()
+    checkLogin()
+
+    for (element in responseData) {  // For every element sets its value in HTML
+        const elementField = document.getElementById(element)
+        if (elementField) {
+            if (element === "extra_ids") {
+                elementField.value = responseData[element]
+            } else {
+                elementField.placeholder = responseData[element]
+            }
+        }
+    }
+}
+
+const renderClientSidePage = async () => {  // Makes the GET request and calls fetchResponseToHtml with response as arg
+    const response = await fetch(requestUrl).catch(err => { })
+    await fetchResponseToHtml(response);
+}
+
+const updateSettings = () => {  // Creates a JSON with the updated settings, if a setting is not updated fills it with the old one
+    requestJson = {}
+    fields.forEach(field => {
+        requestJson[field] = document.getElementById(field).value ? document.getElementById(field).value : document.getElementById(field).placeholder
+    })
+    requestJson["extra_ids"] = [...normalizeExtraIds(requestJson["extra_ids"])]
+    postUpdateRequest(requestJson)
+}
+
+const updateStatusBar = (res) => {   
+    const statusBar = document.getElementById("status_bar")
+    res.status === 200 ? statusBar.innerText = "Success!" : statusBar.innerText = "Failure, please try again!"
+}
+
+const postUpdateRequest = async (payload) => {  // POSTs the request for updating settings
+    const res = await fetch(requestUrl, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json'
+          },
+        body: JSON.stringify(payload)
+    })
+    updateStatusBar(res)
+    renderClientSidePage()
+} 
+
+const deleteMedia = async () => {
+    const deleteMediaUrl = `${baseUrl}/${API_PREFIX}/delete-media/`
+    const res = await fetch(deleteMediaUrl)
+    renderClientSidePage()
+}
+
+const logout = async () => {
+    const logoutUrl = `${baseUrl}/${API_PREFIX}/logout/`
+    const res = await fetch(logoutUrl)
+    renderClientSidePage()
+}
+
+
+
+const submitBtnListener = () => {  // Event listener for the submit button
+    const submitBtn = document.getElementById("submitBtn")
+    submitBtn.addEventListener("click", () => {
+        updateSettings()
+    })
+}
+
+const deleteMediaBtnListener = () => {
+    const deleteMediaBtn = document.getElementById("deleteMediaBtn")
+    deleteMediaBtn.addEventListener("click", () => {
+        if(confirm("Are you sure you want to delete all medias?")) {
+            deleteMedia()
+        }
+    })
+}
+
+const logoutBtnListener = () => {
+    const logoutBtn = document.getElementById("logoutBtn")
+    logoutBtn.addEventListener("click", () => {
+        if(confirm("Are you sure you want to logout?")) {
+            logout()
+        }
+    })
+}
```

Comment:
Just create a single function, no reason to have 3 different one to attachEvent listener